targets/pano_logic_g2/base.py

```diff
@@ -73,7 +73,32 @@ def __init__(self, platform, **kwargs):
                 interface=self.cpu.debug_bus,
                 size=0x100)
 
-        # ??????
+        # Memory mapped SPI Flash ------------------------------------------------------------------
+        self.submodules.spiflash = spi_flash.SpiFlash(
+            platform.request("spiflash"),
+            dummy=platform.spiflash_read_dummy_bits,
+            div=platform.spiflash_clock_div,
+            endianness=self.cpu.endianness)
+        self.add_csr("spiflash")
+        self.add_constant("SPIFLASH_PAGE_SIZE", platform.spiflash_page_size)
+        self.add_constant("SPIFLASH_SECTOR_SIZE", platform.spiflash_sector_size)
+        self.add_constant("SPIFLASH_TOTAL_SIZE", platform.spiflash_total_size)
+        self.add_wb_slave(
```

Comment:
Maybe instead of using `add_wb_slave` and `add_memory_region` you could just call `register_mem` just like in https://github.com/timvideos/litex-buildenv/blob/master/targets/icefun/base.py#L78?